src/services/cart.js

```diff
@@ -0,0 +1,111 @@
+const  { Cart, CartItem  }= require('../config');
+
+class CartService{
+    async createCart (userId){
+        //Create the cart adding the User id as FK
+        console.log('Id', userId);
+        const cart =Cart.build({ 
+            status: 'new',
+            total: 0,
+            date: new Date(),
+            UserId: userId
+        });
+        await cart.save();
+        return cart
+    }
+    
+    async findAllCarts(){
+        const carts = await Cart.findAll();
+        if (carts===null){
+            throw new Error('There are no existing Carts');
+        }else{
+            return carts;
+        }
+    }
+    
+    async findCartbyId(id){
+        //find the Cart by his own Id
+        const cart = await Cart.findByPk(id,{ include: [CartItem] });
+        if (cart===null){
+            throw new Error('Cart not found');
+        }else{
+            return cart;
+        }
+    }
+
+    async findCartbyUserId(userData) {
+        //Find the Cart by the User Id that owns the cart. 
+        const cart = await Cart.findOne({where: { UserId: userData.UserId}});  
+        if (cart===null){
+            throw new Error('User cart not found');
+        }else{
+            const data = await this.findCartbyId(cart.id);
+            return data;
+        }
```

Comment:
I think we are doing the same query twice here. 

The result of `cart` and `data` are the same because the userId would be related to only one cart.

Reply:
Made the correction and now the:

`const cart = await Cart.findOne({where: { UserId: userData.UserId}}); ` now is
`const cart = await Cart.findOne({where: { UserId: userData.UserId}, include: [CartItem] });`

the difference was that the findCartById( ) function i made has the findbyPk() with the `include:[CartItem]` cause i thought the findOne didn't had the include too, but now i made the try and it works.
So now we have the possibility to find the Cart by Id or the UserId, the endpoint makes use of the UserId one.

Reply:
Awesome! that's a really good improvemnt! Thank you!